### ISSUE TITLE
Deduplicate "comma and 'and'" kata

### ISSUE THREAD

**Issue by hobovsky** [codewars/content-issues#58]
1. [Comma, comma and and](https://www.codewars.com/kata/565c88c862901d1ad9000074)
- 7 kyu
- No Oxford comma
- 184 solutions, 93% satisfaction
- No pending issues
- Ruby, pending JS translation
- author active a month ago
- published Nov 2015

2. [The Oxford Comma](https://www.codewars.com/kata/540d08313b653278c3000b11)
- 6 kyu
- Requires Oxford comma (duh)
- 55 solutions, 93% satisfaction
- Pending issues about missing random tests and broken sample tests
- Coffeescript only(!), no pending translations
- author active a year ago
- published Sep 2014

3. [Format words into a sentence](https://www.codewars.com/kata/51689e27fe9a00b126000004)
- 6 kyu
- No Oxford comma
- 3,578 solutions, 88% satisfaction
- Pending issues about missing random tests in some languages
- 7 languages, no pending translations
- authored by jhoffner
- older than Codewars itself, published Apr 2013

4. [Format a string of names like 'Bart, Lisa & Maggie'.](https://www.codewars.com/kata/53368a47e38700bd8300030d)
- 6 kyu
- A bit more complex input, slightly different output format
- 71211 solutions(!), 88% satisfaction
- 4 pending issues, but some of them probably obsolete
- 3.5 languages + 7 pending translations
- author inactive
- published Mar 2014

I think that missing Oxford comma in 1. and 3. makes the requirements not really correct, but I am no language expert so maybe it would be good to have this confirmed/denied by English speakers.
6 kyus seem to be a tad overranked for today's standards, but reranking them will be difficult due to large amount of solutions.

------

Conclusion:

- [x] Retire [Comma, comma and and](https://www.codewars.com/kata/565c88c862901d1ad9000074)
- [x] Retire [The Oxford Comma](https://www.codewars.com/kata/540d08313b653278c3000b11)
- [x] Retire [Format a string of names like 'Bart, Lisa & Maggie'.](https://www.codewars.com/kata/53368a47e38700bd8300030d) (note: over 70k solutions)
- Keep [Format words into a sentence](https://www.codewars.com/kata/51689e27fe9a00b126000004)

**Comment by hobovsky** [comment]
My vote:

- If the lack of the Oxford comma is not a critical issue, keep 3. as it has the most languages available, and retire the rest. Make sure that the note about missing Oxford comma stands out prominently enough to prevent issues about invalid grammar.
- ~~If grammar is a concern here, keep 4. since it's not affected by grammar issues (or is it?), it has good selection of languages, and large amount of solutions. Retire the rest.~~ On a second thought, I don't like the input format here. I change my alternative vote to: if the Oxford comma is a critical thing, keep 2. and provide missing translations, retire the rest.
- Since  reranking might not be an option, consider hardening the tests in a way they reject solutions based on concatenation of immutable strings in a loop (for applicable languages).

**Comment by awesomeAD1** [comment]
1. is Ruby only, as a duplicate kata it should have lowest survival priority.
2. The requirement of an Oxford comma actually simplifies the implementation, as far as I'm concerned, which makes this kata _less_ interesting.
3. Most translations, no issues. While not highest solution rate, I don't think that should be given too much weight; after all, older kata simply have more solutions, but removing duplicates and fixing kata is a forward-looking process.
4. The input format merely makes extracing the relevant strings more complex than it needs to be and doesn't actually enhance the core exercise of the kata.

TL;DR: keep 3.

EDIT:

Just to clarify my view on the issue of the Oxford comma from a grammatical standpoint: 
I am not aware that the Oxford comma is a grammatical _rule_ as much as a recommendation. Personally, I view it more as a helpful tool to distinguish those cases where the last item in a comma-separated list is itself comprised of multiple, and-separated things. I take its absence as a sign that this isn't the case.
To discourage complaints and discussions from pedantics in the discourse, a note could be added to the description that acknowledges the disregard of the Oxford comma as a conscious choice.

**Comment by scoraig52** [comment]
I go with 3 for aforementioned reasons.

**Comment by Kacarott** [comment]
I also vote for keeping 3, additionally for historical reasons.

**Comment by ejini6969** [comment]
i) 1. is a duplicate of 3. So it should definitely be retired
ii) 2. has only CS enabled and addition of Oxford comma does not make it unique enough from other tasks (plus the usage of Oxford comma in my country is actually considered as grammatically incorrect, at least that's what we were thought in primary schools and when writing essays, but I agree that is just a matter of taste rather than a rigid rule)
iii) As for 4. I don't like the input, it adds additional noise when translating with the description being a mess (hard to make it to language agnostic and does not add much context to the task, an **array of strings** should just be fine enough)

Final decision is to keep 3. due to reasons mentioned by others and above.

**Comment by kazk** [comment]
> older than Codewars itself, published Apr 2013

Codewars was created in 2012. It's slmost 10 years old.
